File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right = 0, 1
        profit = 0  
        maxProfit = 0 

        while right < len(prices):
            if prices[left] < prices[right]:
                profit = max(profit, prices[right] - prices[left])
                maxProfit = profit if profit > maxProfit else maxProfit
            else:
                left = right
            right += 1
        
        return maxProfit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = left + 1;
        int res = 0;
        while (right < prices.length) {
            if (prices[left] < prices[right]) {
                res = Math.max(res, prices[right] - prices[left]);    
            }
            else {
                left = right;
            }
            right++;

        }
        return res;
    }   
}
